server/plugins/client/bundle.js

```diff
@@ -14,20 +28,27 @@ var parallel = require('async').parallel
  * and avoiding unneeded bundling with browserify saves a significant time.
  */
 function bundleClient (hoodieClientPath, bundleTargetPath, config, callback) {
-  parallel([
+  var plugins = [
+    path.resolve(config.paths && config.paths.applicationRoot || '', 'hoodie/client')
+  ].filter(checkModule)
+  var getPluginsModifiedTimes = plugins.map(function (pluginPath) {
+    return getModifiedTime.bind(null, requireResolve(pluginPath))
+  })
```

Comment:
> I wouldn’t mind to leave things simple as they are right now and document its limitation, with a link to an issue that describes it in detail and what the approach would be to fix it, so that we can work on this separately.

As you correctly interpret above, this code will get the path to the entry file of the client code, so we can place in the `README.md` section regarding how to write code for an app that is consumed by browserify and incorporated into the bundle ([nts](# "note to self"): which I should write as a final step for this PR) that you need to `touch` or otherwise alter the mtime on the entry file for the client code for it to be reconsumed, with a link to the issue-to-be-created you mention.

I don't think this is too much for someone at the moment, especially as the bundle is only re-built on a server restart for now due to the clever cache-by-Promise mechanism in `server/plugins/client/bundle-handler-factory.js`

> A few more thoughts for the bundled hoodie.js client: [...]

I think it's worth raising these as separate issues in addition to the above one, to at least help categorise thoughts on the separate topics.